ecl/regress/volatile1.ecl

```diff
@@ -0,0 +1,50 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2016 HPCC Systems®.
+
+    All rights reserved. This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+############################################################################## */
+
+
+// The following assume RANDOM() aren't strictly correct, but assume it is unlikely for RANDOM() to return the
+// same number twice in succession.
+
+//Random should be re-evaluated for each unique instance.
+output(IF(random() != random(), 'Pass', 'Fail'));
+
+//There is only a single instance of this variable - it should not be re-evaluated
+volatile1 := random();
+output(IF(volatile1 = volatile1, 'Pass', 'Fail'));
+
+//Again, re-evaluating the function should not create a new value
+volatile2() := random();
+output(IF(volatile2() = volatile2(), 'Pass', 'Fail'));
+
+//Again, no reason for the random in the function to be re-evaluated.
+volatile3(integer x) := random() % x;
+output(IF(volatile3(100) % 50 = volatile3(50), 'Pass', 'Fail'));
+
```

Comment:
It is debatable.  If the use of random() inside a function creates a new instance for each combination of parameters, then it can get very confusing.  That is what the code did before and it ends up causing lots of extra instances of random(), and prevents calls to the same function being commoned up.